synapse/handlers/directory.py

```diff
@@ -84,6 +84,13 @@ def get_association(self, room_alias):
             if result:
                 room_id = result.room_id
                 servers = result.servers
+            else:
+                # Query AS to see if it exists
+                as_handler = self.hs.get_handlers().appservice_handler
+                result = yield as_handler.query_room_alias_exists(room_alias)
+                if result:
+                    room_id = result.room_id
+                    servers = result.servers
```

Comment:
Fixed in https://github.com/matrix-org/synapse/commit/e9484d6a9514d5c9aea3839bcc3fb25c9cb88cb0